database/seed.js

```diff
@@ -0,0 +1,164 @@
+var Promise = require("bluebird");
+var faker = require('faker');
+var db = require('./dBHelpers.js')
+let mysql = require('mysql');
+let connection = mysql.createConnection({
+  host: "localhost",
+  user: "root",
+  password: "",
+  database: "product"
+});
+
+
+
+// Seed a product and many color/size relationships
+// Call this seed 100 times
+
+
+
+// Generate a product with:
+var createProduct = function() {
+  console.log('create product is running')
+  var product = {}
+  // name
+  product.productName = faker.commerce.productName();
+  // brand
+  product.productBrand = faker.company.companyName();
+  // price
+  product.productPrice = faker.commerce.price();
+
+
+
+
+  //////////////////////////////////////////////////////////
+  var productSeed = new Promise((resolve, reject) => {
+    console.log('inside promise');
+
+    let queryString = `INSERT INTO products(product_name, product_brand, product_price) VALUES(?,?,?)`;
+
+    // assign proper values:
+    // let values = [productDetails.productName, productDetails.productBrand, productDetails.productPrice];
+    let values = [product.productName, product.productBrand, product.productPrice];
+
+
+    connection.query(queryString, values, (err, results, fields) => {
+      if (err) {
+        console.error(err.message);
+        //////////////////////////
+        reject(err);
+      }
+      // get inserted id
+      console.log('Product Id:' + results.insertId);
+      ///////////////////////////
+      // resolve(results.insertId);
+      resolve(results.insertId);
+    });
+
+    // return db.insertProduct(product);
+  });
+
+
+  productSeed.then((productId) => {
+
+      var colors = [];
+      var id = 1;
+      // For each product, generate:
+      var numberOfColors = getRandomInt(1, 7);
+        // 1-6 colors - with a picture for each
+      for (var i = 0; i < numberOfColors; i++) {
+        var newColor = generateColor();
+
+        /////////////////////////////////////////////////////////
+        newColor.colorId = id;
+        id++
+        db.insertColor(productId, newColor);
+        colors.push(newColor);
+      }
```

Comment:
You could move these lines into the generateColor function and refactor so that it's similar to your generateSizes and generateAssociations for better consistency.